### DISCUSSION TITLE
Installation checker discussion

### DISCUSSION THREAD

**Discussion by set-soft** [INTI-CMNB/KiBot#209] [KiBot check]
This is open to collect opinions about the installation checker tool: `kibot-check`

Did it help? Did it said something confusing? Something was wrongly detected? etc.

**Comment by dbriscoe511** [comment]
Seems useful as a thing to paste in bug reports

**Comment by erikbozelie** [comment]
Hi, 

I am running manjaro(5.18.10-1-MANJARO) and running into two problems with kibot-check

One is being it is saying some tools are not installed or too old
These ones are:
* InteractiveHtmlBom
* Ghostscript
* RAR

The other problem with kibot is that it cant use certain tools
ERROR:Failed to run /usr/bin/rsvg-convert, error 101 (kibot - out_navigate_results.py:136)
ERROR:Failed to run /usr/bin/convert, error 1 (kibot - out_navigate_results.py:136)

All of the above has been installed. I can run for example gs, gsc, rsvg-convert or convert  from a commandline.
All versions of the installed tools are the latest.

**Reply by set-soft** [reply]
Hi @erikbozelie !
- InteractiveHtmlBom: try using:
```
export INTERACTIVE_HTML_BOM_NO_DISPLAY=True
```
If iBoM can't be detected try running: `generate_interactive_bom.py --version`

- Ghostscript: Do you have the `ghostscript` command? This is the more descriptive modern version, Debian and Ubuntu has it, perhaps we should use the old cryptic `gs` name.
- RAR: What do you get running `rar -?`
- About the other errors: try running KiBot with debug info, lets say `-vvv`, then look at the log and see if you have the output from the tools to identify the problem. BTW did you check if ImageMagick policy allows manipulating PDFs? (mentioned [here](https://github.com/INTI-CMNB/KiBot/#dependencies))

**Reply by erikbozelie** [reply]
Hi Set-soft,

Thank you for your reply, ill try to go over it one by one.

InteractiveHtmlBom:
This one is installed: https://aur.archlinux.org/packages/kicad-interactivehtmlbom-git (v2.3.something)
Kicad-check give me:
Interactive HTML BoM: Git version check failed: Command '['git', 'describe', '--tags', '--abbrev=4', '--dirty=-*']' returned non-zero exit status 128.

summary:
* Interactive HTML BoM not installed or too old 
 Visit: https://github.com/INTI-CMNB/InteractiveHtmlBom                                                                                                                                     
 Download it from: https://github.com/INTI-CMNB/InteractiveHtmlBom/releases                                                                                                                 
 - Mandatory for `ibom`  

Unstalled the 2.3 version from aur and tried to upgrade:

Attempt1: Ive downloaded https://github.com/INTI-CMNB/InteractiveHtmlBom/releases/download/v2.5.0/interactivehtmlbom.inti-cmnb_2.5.0-1_all.deb
Converted it into something i can use
“deptab [interactivehtmlbom.inti-cmnb_2.5.0-1_all.deb](https://github.com/INTI-CMNB/InteractiveHtmlBom/releases/download/v2.5.0/interactivehtmlbom.inti-cmnb_2.5.0-1_all.deb)”
then tried to install, but its asking for kicad-nightly

Attempt2: i installed the 2.5 version via the plugin manager of kicad itself. This worked. But now kicad-check gives me: 

“Interactive HTML BoM: *UNKNOWN*”

With same message in summary

Attempt3: Install a difference version from aur, same result as attempt 2

Ghostscript:
The default install from the official repository(extra) installs this under the command gsc with gs being a symlink.
Ive added a symlink with the command below which fixes the errors in kibot-check
```sudo ln -s gsc ghostscript ```in /usr/bin

Rar:
Ive fixed the warning with the same symlink command as ghostscript
```sudo ln -s unrar rar``` in /usr/bin
Dont know if the commands are exactly the same. Ill do some testing with rar.


Other Errors:

Ive looked into /etc/ImageMagick-7/policy.xml.
Its mostly empty except for “
2 header taggs, Lots of comments 
and
```
<policymap>
  <policy domain="Undefined" rights="none"/>
</policymap>”
```
There are allot of steps stopping the script from executing.
These modules are 
“
costs_bom_xlsx
basic_gencad
classic_netlist
basic_pcb_print_svg
basic_pcb_print_png
basic_pcb_print_eps
basic_pcb_print_ps

basic_pcbdraw_svg_jlcpcb_green_enig_top(all variants)

basic_pdf_sch_print
basic_render_3d_top
basic_render_3d_30deg
basic_render_3d_bottom
basic_step
basic_svg_sch_print”

A different project has some different modules.

Thanks in advance,
Erik

**Reply by set-soft** [reply]
Hi @erikbozelie !

> InteractiveHtmlBom: This one is installed: https://aur.archlinux.org/packages/kicad-interactivehtmlbom-git (v2.3.something) Kicad-check give me: Interactive HTML BoM: Git version check failed: Command '['git', 'describe', '--tags', '--abbrev=4', '--dirty=-*']' returned non-zero exit status 128.

The `Git version check failed: Command '['git', 'describe', '--tags', '--abbrev=4', '--dirty=-*']' returned non-zero exit status 128.` is normal when using the original iBoM (not patched). This is something I should fix with a PR. 

> * Interactive HTML BoM not installed or too old

Which is true, even if kibot-check could detect the version, you need at least 2.4.1-4

> Unstalled the 2.3 version from aur and tried to upgrade:
> 
> Attempt1: Ive downloaded https://github.com/INTI-CMNB/InteractiveHtmlBom/releases/download/v2.5.0/interactivehtmlbom.inti-cmnb_2.5.0-1_all.deb Converted it into something i can use “deptab [interactivehtmlbom.inti-cmnb_2.5.0-1_all.deb](https://github.com/INTI-CMNB/InteractiveHtmlBom/releases/download/v2.5.0/interactivehtmlbom.inti-cmnb_2.5.0-1_all.deb)” then tried to install, but its asking for kicad-nightly

The dependency is `kicad (>= 5.1.0) | kicad-nightly` so what you really need is `kicad`

> Attempt2: i installed the 2.5 version via the plugin manager of kicad itself. 

This will have the same issue about trying to run `git` to detect the git hash.

> This worked. But now kicad-check gives me:
> 
> “Interactive HTML BoM: _UNKNOWN_”
> 
> With same message in summary

Sounds logic.


> Attempt3: Install a difference version from aur, same result as attempt 2

I guess this was a stock iBoM.

Ok, I'll see what can be done for cases where you can't install the .deb file. I didn't notice it was that hard without using the .deb file.

> Ghostscript: The default install from the official repository(extra) installs this under the command gsc with gs being a symlink. Ive added a symlink with the command below which fixes the errors in kibot-check `sudo ln -s gsc ghostscript `in /usr/bin

Ok, looks like `gs` is a more common name.

> Rar: Ive fixed the warning with the same symlink command as ghostscript `sudo ln -s unrar rar` in /usr/bin Dont know if the commands are exactly the same. Ill do some testing with rar.

On my system unrar is:
```
UNRAR 6.00 freeware      Copyright (c) 1993-2020 Alexander Roshal
```
Without compression options, and rar is:
```
RAR 5.50   Copyright (c) 1993-2017 Alexander Roshal   11 Aug 2017
```
It is from the non-free repo and isn't freeware, is shareware.

 
> Other Errors:

Did you try running with debug information to see if you can get the output from `/usr/bin/rsvg-convert` and `/usr/bin/convert`?

Do you have experience creating docker images with manjaro as base?

**Reply by erikbozelie** [reply]
Hi @set-soft 

Is there a way to let kibot run over / continue with errors so i can send a complete log?
this is what shows up in a log

example hangs on the example from kibot --quick-start
```
- 'Costs Bill of Materials in XLSX format' (costs_bom_xlsx) [bom]
DEBUG:Output destination: /home/PROJECT/BoM/Costs (kibot - kiplot.py:360)
DEBUG:Filters reset (kibot - fil_base.py:161)
DEBUG:Applying transform filter `_kicost_rename,_rot_footprint` (kibot - fil_base.py:140)
DEBUG:Applying filter `_kicost_dnp` to fitted (kibot - fil_base.py:171)
DEBUG:Applying KiCost style variant `place_holder` (kibot - var_kicost.py:75)
DEBUG:Using auxiliary origin: x=0 y=0 (kibot - bom.py:477)
DEBUG:Saving BOM File: /home/erik/PROJECT/BoM/Costs/PROJECT.0-bom.xlsx (kibot - bom.py:548)
Traceback (most recent call last):
  File "/usr/bin/kibot", line 16, in <module>
    main()
  File "/usr/lib/python3.10/site-packages/kibot/__main__.py", line 323, in main
    generate_outputs(outputs, args.target, args.invert_sel, args.skip_pre, args.cli_order, args.no_priority)
  File "/usr/lib/python3.10/site-packages/kibot/kiplot.py", line 470, in generate_outputs
    run_output(out, dont_stop)
  File "/usr/lib/python3.10/site-packages/kibot/kiplot.py", line 397, in run_output
    out.run(get_output_dir(out.dir, out))
  File "/usr/lib/python3.10/site-packages/kibot/out_base.py", line 162, in run
    self.options.run(self.expand_filename(output_dir, self.options.output))
  File "/usr/lib/python3.10/site-packages/kibot/out_bom.py", line 731, in run
    do_bom(output, format, comps, self)
  File "/usr/lib/python3.10/site-packages/kibot/bom/bom.py", line 559, in do_bom
    write_bom(file_name, ext, groups, cfg.columns, cfg)
  File "/usr/lib/python3.10/site-packages/kibot/bom/bom_writer.py", line 46, in write_bom
    result = write_xlsx(filename, groups, headings, head_names, cfg)
  File "/usr/lib/python3.10/site-packages/kibot/bom/xlsx_writer.py", line 825, in write_xlsx
    kicost_colors = create_kicost_sheet(workbook, groups, image_data, fmt_title, fmt_info, fmt_subtitle, fmt_head,
  File "/usr/lib/python3.10/site-packages/kibot/bom/xlsx_writer.py", line 695, in create_kicost_sheet
    return _create_kicost_sheet(workbook, groups, image_data, fmt_title, fmt_info, fmt_subtitle, fmt_head, fmt_cols, cfg)
  File "/usr/lib/python3.10/site-packages/kibot/bom/xlsx_writer.py", line 582, in _create_kicost_sheet
    configure_kicost_apis(cfg.xlsx.kicost_config, True, dis_enable_apis, cfg)
  File "/usr/lib/python3.10/site-packages/kicost/__main__.py", line 175, in configure_kicost_apis
    configure_apis(api_options)
  File "/usr/lib/python3.10/site-packages/kicost/distributors/__init__.py", line 90, in configure_apis
    distributor_class.configure_apis(options)
  File "/usr/lib/python3.10/site-packages/kicost/distributors/distributor.py", line 222, in configure_apis
    api.configure(ops)
  File "/usr/lib/python3.10/site-packages/kicost/distributors/api_digikey.py", line 113, in configure
    format(api_digikey.enabled, DK_API.id, DK_API.secret, cache_path))
AttributeError: type object 'DK_API' has no attribute 'id'
```
When ive removed the sections in the yaml file this is what it shows:
```
DEBUG:- Executing: ['/usr/bin/rsvg-convert', '-d', '72', '-p', '72', '-f', 'pdf', '-o', '/home/erik/PCB/PDF/01/PROJECT.0.pdf', '/home/erik/PCB/PDF/01/PROJECT.0.svg'] (kibot - out_pcb_print.py:59)
ERROR:Failed to run /usr/bin/rsvg-convert, error 101 (kibot - out_pcb_print.py:63)
```

Thank you for all the help so far.
from what i have seen so far Kibot is amazing ^,^

ill be in a conference for a few days and will be a bit later to respond.

**Reply by set-soft** [reply]
Hi @erikbozelie !
 
> Is there a way to let kibot run over / continue with errors so i can send a complete log? this is what shows up in a log

The `--quick-start` does it. I added a new command line option to set the dont_stop flag https://github.com/INTI-CMNB/KiBot/commit/95a48a892244038caca6e192695af8e7ff81c5e5

> 
> example hangs on the example from kibot --quick-start
> 
> ```
> - 'Costs Bill of Materials in XLSX format' (costs_bom_xlsx) [bom]
> DEBUG:Output destination: /home/PROJECT/BoM/Costs (kibot - kiplot.py:360)
> ...
>   File "/usr/lib/python3.10/site-packages/kicost/distributors/api_digikey.py", line 113, in configure
>     format(api_digikey.enabled, DK_API.id, DK_API.secret, cache_path))
> AttributeError: type object 'DK_API' has no attribute 'id'
> ```

Ok, didn't notice it, I added a conditional to avoid this problem when the plug-in isn't installed.  https://github.com/hildogjr/KiCost/commit/ac0d58277c30a6e35064dc784bce5cc079632cf8

> 
> When ive removed the sections in the yaml file this is what it shows:
> 
> ```
> DEBUG:- Executing: ['/usr/bin/rsvg-convert', '-d', '72', '-p', '72', '-f', 'pdf', '-o', '/home/erik/PCB/PDF/01/PROJECT.0.pdf', '/home/erik/PCB/PDF/01/PROJECT.0.svg'] (kibot - out_pcb_print.py:59)
> ERROR:Failed to run /usr/bin/rsvg-convert, error 101 (kibot - out_pcb_print.py:63)
> ```

No `Output from command` message? Can you try running:
```
/usr/bin/rsvg-convert -d 72 -p 72 -f pdf -o test.pdf AN_SVG_GENERATED_BY_KICAD
```
All I can find about 101 error level is related to network issues

> Thank you for all the help so far. from what i have seen so far Kibot is amazing ^,^
:-)

> ill be in a conference for a few days and will be a bit later to respond.
No problem.

**Reply by erikbozelie** [reply]
Thank you for the “dont stop flag”, ive rebuilded kibot and that helps allot.

Ive unstalled kicost from aur via the package manager and installed the pip version.
This fixed the crashing issue for this module

```
pip install kicost
```

Ive also ran the command below and it works as expected
```
/usr/bin/rsvg-convert -d 72 -p 72 -f pdf -o test.pdf Project_V2.0-B_Courtyard.svg

```

A file named test.pdf is generated with expected contents

Ive exported my entire verbose log to a file and shared it here:
https://drive.google.com/file/d/1Au47n2W3s-qdtQ67FT1wyVMd0Av0B9NW/view?usp=sharing

There are a few errors left but its getting there.

**Reply by set-soft** [reply]
Hi @erikbozelie !
Looking at the log I can see:

> FileNotFoundError: [Errno 2] No such file or directory: 'recordmydesktop'

This is an optional dependency needed when using debug, is documented. I'm adding to my TODO a better handling for this situation. Just install it.

> DEBUG:- Executing: ['/usr/bin/rsvg-convert', '-d', '72', '-p', '72', '-f', 'pdf', '-o', '/home/erik/kibot_test/Generated/PCB/PDF/01/project.pdf', '/home/erik/kibot_test/Generated/PCB/PDF/01/project.svg'] (kibot - out_pcb_print.py:74)
ERROR:Failed to run /usr/bin/rsvg-convert, error 101 (kibot - out_pcb_print.py:78)
DEBUG:Output from command: thread 'main' panicked at 'assertion failed: h >= 0.0', src/text.rs:360:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
 (kibot - out_pcb_print.py:80)

The problem seems to be related to `/home/erik/kibot_test/Generated/PCB/PDF/01/project.svg`. Version 2.50.3 doesn't have any problem on my system. You can try:

```
$ export RUST_BACKTRACE=1
$ /usr/bin/rsvg-convert -d 72 -p 72 -f pdf -o /home/erik/kibot_test/Generated/PCB/PDF/01/project.pdf /home/erik/kibot_test/Generated/PCB/PDF/01/project.svg
```

And see if it gives more information about the problem, but I'm not sure. Can you post  `/home/erik/kibot_test/Generated/PCB/PDF/01/project.svg` to check it can be converted on my system?

> DEBUG:Executing ['/home/erik/.local/bin/pandoc', '--from', 'markdown', '--resource-path=/home/erik/kibot_test/Generated', '/home/erik/kibot_test/Generated/project-report.txt', '-o', '/home/erik/kibot_test/Generated/project-report.pdf'] (kibot - out_report.py:741)
ERROR:/home/erik/.local/bin/pandoc error: 43 (kibot - out_report.py:745)
DEBUG:Output from command: [WARNING] Could not fetch resource 'Schematic/project-schematic.svg': replacing image with description

This problem should go away when installing recordmydesktop. 

> thread 'main' panicked at 'assertion failed: h >= 0.0', src/text.rs:360:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
[WARNING] Could not convert image './tex2pdf.-053bebc04fff435c/a46e253ac379c219323f3f47fd05e9e969ea7e38.svg': conversion from SVG failed
> (process:33954): Pango-WARNING **: 10:25:58.466: failed to create cairo scaled font, expect ugly output. the offending font is 'Liberation Serif 1333333'
> (process:33954): Pango-WARNING **: 10:25:58.466: font_face status is: error occurred in libfreetype
> (process:33954): Pango-WARNING **: 10:25:58.466: scaled_font status is: error occurred in libfreetype

And here PanDoc hits the same problem we found with `rsvg-convert`. But here Pango says that `Liberation Serif` font has some problem.

As the file can't be converted to PDF then Ghostscript fails:

> ! LaTeX Error: Unknown graphics extension: .svg.

And here is the `convert` issue:

> DEBUG:- Executing: ['/usr/bin/convert', '/home/erik/kibot_test/Generated/PCB/PNG/project-assembly_page_01.png', '-resize', '256x', '/home/erik/kibot_test/Generated/Browse/images/file_png_64.png', '-gravity', 'south-east', '-composite', '/home/erik/kibot_test/Generated/Browse/images/cat_basic_pcb_print_png_project-assembly_page_01.png'] (kibot - out_navigate_results.py:136)
> ERROR:Failed to run /usr/bin/convert, error 1 (kibot - out_navigate_results.py:140)
> DEBUG:Output from command: convert: improper image header `/home/erik/kibot_test/Generated/PCB/PNG/project-assembly_page_01.png' @ error/png.c/ReadPNGImage/4303.
convert: image sequence is required `-composite' @ error/mogrify.c/MogrifyImageList/8187.
convert: no images defined `/home/erik/kibot_test/Generated/Browse/images/cat_basic_pcb_print_png_project-assembly_page_01.png' @ error/convert.c/ConvertImageCommand/3325.
 (kibot - out_navigate_results.py:142)

Perhaps this file is empty/broken because it was a failed conversion from SVG to PNG. 
Other errors seems to be chained with rsvg-convert failing and/or producing broken PNG files.

Post one of the problematic SVG files to see if the problem is in the file or is just something related to the fonts in your system.

**Reply by set-soft** [reply]
Hi @erikbozelie !
Any news about the problem with the SVG files?

**Reply by erikbozelie** [reply]
> Hi @erikbozelie ! Any news about the problem with the SVG files?

Hi Set-soft,

Sorry for the slow response.
Ive done a complete reinstall to see if i did something wrong. I have installed kibot  on a local server (running the same kernel, OS etc) to see if i messed something up with my installation.
I was writing a report but got pulled off it.

From my memory: Interactive bom is working but kibot-check is reporting that it cant see the git version.
Same problem as before with the SVG modules. 

Ill come back to you on monday when in back at the office.

Have a nice weekend,
Erik

**Reply by erikbozelie** [reply]
Hi @set-soft 

I cant give you my work related files, so Ive created a small addon pcb for a badge and ran it through kibot(hobby).
https://drive.google.com/file/d/1FP0mYTO_7lgzHmQ30ogchrIz3cdCQOx0/view?usp=sharing

First run is just having kibot installed
Second run has everything but Rar

Ive installed kibot on a fresh installation of manjaro to avoid me having to have messed something up.

```
>which rsvg-convert                                                                                                                                                                                                                          
/usr/bin/rsvg-convert
>which pcbdraw                                                                                                                                                                                                                               
/usr/bin/pcbdraw
>pcbdraw --version                                                                                                                                                                                                                           
PcbDraw 0.9.0-2+21.g415ce87
>rsvg-convert --version                                                                                                                                                                                                                      
rsvg-convert version 2.54.4
```

The git error from the interactive html bom is below, but appears to be working.
```
Interactive HTML BoM: Git version check failed: Command '['git', 'describe', '--tags', '--abbrev=4', '--dirty=-*']' returned non-zero exit status 128.
```

Can pandoc be the problem? Latest version in the repository is 2.18. i have 2.5 installed via manual means.

**Reply by set-soft** [reply]
Hi @erikbozelie !

Thanks for the logs.

> Ive installed kibot on a fresh installation of manjaro to avoid me having to have messed something up.

Do you have a list of commands to do it? I could try to reproduce your environment using docker.

> The git error from the interactive html bom is below, but appears to be working.

Yes, this error is missleading, I asked about it to the author (https://github.com/openscopeproject/InteractiveHtmlBom/issues/321)

> Can pandoc be the problem? Latest version in the repository is 2.18. i have 2.5 installed via manual means.

Don't know, the problem seems to be related to the fonts, I don't know if pandoc is pulling some fonts dependencies on Manjaro.

The KiAuto you are trying is old, but I'm not sure about the reason, the logs says:

```
DEBUG:- Trying to download https://api.github.com/repos/INTI-CMNB/KiAuto/releases/latest (kibot - dep_downloader.py:123)
DEBUG:- Trying to download https://api.github.com/repos/INTI-CMNB/KiAuto/tarball/v1.6.15 (kibot - dep_downloader.py:123)
```

But currently latest is 2.0.0, so this log seems to have more than 5 days. This release should solve the recordmydesktop issue (https://github.com/INTI-CMNB/KiAuto/commit/f98fa7072cc6ce58dc67e497062f792b73ff40f9).

In the first run KiAuto doesn't work because `psutil` isn't installed.

A lot of stuff fails to install in the first run when using pip, lamentably the code failed to log the error. I'm fixing the error capture.

The PcbDraw you have: 0.9.0.post2+21.g415ce87 isn't functional. I saw complains in the upstream site. So now I'm boosting the dependency to 0.9.0.3, which is my fork. As a consequence of using the broken PcbDraw a lot of things weren't generated.

I suggest upgrading KiAuto to 2.0.0 and installing PcbDraw from the last release: https://github.com/INTI-CMNB/PcbDraw/releases/tag/v0.9.0-3 in the fresh setup, to see if it fixes the problems.

Trying the last KiBot patches (originated from this discussion) could also help.

**Reply by erikbozelie** [reply]
Hi @set-soft 

Here is a complete commands i used
This should do most of the installation how i have it:

Upgrading os, installing aur helper and kicad
```
sudo pacman -Syyu
sudo pacman -S yay
sudo pacman -S kicad kicad-library kicad-library-3d
```

installing kibot from git and its dependancy’s
```
yay -S kiauto-git kibom-git kibot-git
```

kibot-check will now be available

Lets fix the dependancy’s
```
yay -S kicost-git qrcodegen
sudo pacman -S pandoc
```


Some of the things needed to be installed are also available through the package manager. These do not seem to work. The ones from pip do.
```
pip install kicost
pip install KiKit
pip install qrcodegen
pip install kiauto
pip install svgpathtools

```
Interactive html bom
```
via its kicad’s gui plugin and content manager
```

reboot

The official repository pcbdraw version is 0.6.1
Can switch to 0.9.0.2 with
```
yay -S pcbdraw-git
```

We should also install that recordmydesktop app
```
sudo pacman -S recordmydesktop
```
Thanks for the help.
Ill try to upgrade kiauto to V2.0 and pcbdraw to version 0.9.0.3 to see if that fixes some issues,
Erik

**Reply by set-soft** [reply]
Hi @erikbozelie !
I was playing with a Manjaro image:
```shell
# Download the current image
docker pull manjarolinux/base
# Run bash inside the image, options:
# Automatically remove the container when it exits
# Interactive + TTY
docker run --rm -it manjarolinux/base /bin/bash
# Use a regular user, this one is always there
su builder
# Use the home as workplace
cd
# Download the script
curl https://raw.githubusercontent.com/INTI-CMNB/KiBot/master/tools/arch_install.sh > arch_install.sh
# Run it
. arch_install.sh
```

The script that I'm currently tring does:

```shell
#!/bin/sh
# Update the system
sudo pacman -Syyu --noconfirm
# Install the yay installer (needed for AUR) (Also installs git)
sudo pacman -S yay binutils --noconfirm
# Python installer
sudo pacman -S python-pip --noconfirm
###############################################################################
# Big dependencies
###############################################################################
# KiCad install, no 3D models (Down: 308.75 MiB, size: 1617.84 MiB)
# python-wxpython is installed here
sudo pacman -S kicad kicad-library --noconfirm
# KiBot graphic deps
# Note: librsvg is already installed for KiCad
sudo pacman -S ghostscript imagemagick librsvg --noconfirm
# Install rar (from AUR KiBot optional)
yay -S --noconfirm rar
# Install pandoc (KiBot optional)
sudo pacman -S pandoc texlive-core --noconfirm
# Not needed at all, already installed and very optional
# sudo pacman -S poppler --noconfirm
###############################################################################
# KiAuto
###############################################################################
# Dependencies
sudo pacman -S recordmydesktop xdotool xclip libxslt python-psutil python-xvfbwrapper --noconfirm
# KiAuto, clean install, no extra packages
sudo pip install KiAuto
###############################################################################
# KiBoM
###############################################################################
# KiBoM & KiBot dependency
sudo pacman -S python-xlsxwriter --noconfirm
# Install KiBoM, clean install, no extra packages (from repo)
git clone https://github.com/INTI-CMNB/KiBoM.git
cd KiBoM/
sudo pip install .
cd ..
sudo rm -r KiBoM/
###############################################################################
# iBoM
###############################################################################
# Install iBoM, clean install, no extra packages
git clone https://github.com/INTI-CMNB/InteractiveHtmlBom.git
cd InteractiveHtmlBom/
sudo pip install .
cd ..
sudo rm -r InteractiveHtmlBom/
###############################################################################
# KiDiff
###############################################################################
# Install KiDiff, clean install, no extra packages
sudo pip install KiDiff
###############################################################################
# PcbDraw 0.9.0
###############################################################################
# Dependencies
sudo pacman -S python-numpy python-lxml python-mistune1 python-pybars3 python-wand python-yaml python-pcbnewtransition python-scipy --noconfirm
yay -S python-svgpathtools-git --noconfirm
# Install PcbDraw, clean install, no extra packages
git clone https://github.com/INTI-CMNB/PcbDraw.git
cd PcbDraw/
git checkout v0.9.0_maintain
sudo pip install .
cd ..
sudo rm -r PcbDraw/
###############################################################################
# KiCost Digi-key plug-in
###############################################################################
# kicost_digikey_api_v3 dependencies
# Repeated: sudo pacman -S python-requests python-urllib3 python-six python-certifi python-setuptools --noconfirm
sudo pacman -S python-inflection python-pyopenssl python-tldextract python-dateutil --noconfirm
# Install Digi-Key KiCost plug-in, clean install, no extra packages
sudo pip install kicost_digikey_api_v3 dependencies
###############################################################################
# KiCost
###############################################################################
# KiCost dependencies
# Repeated: sudo pacman -S python-lxml python-xlsxwriter python-requests python-yaml --noconfirm
sudo pacman -S python-beautifulsoup4 python-tqdm python-validators python-colorama python-pillow --noconfirm
# Install KiCost plug-in, clean install, no extra packages
sudo pip install kicost
###############################################################################
# KiBot
###############################################################################
# KiBot install, pulls qrcodegen dependency
sudo pip install kibot
```

I know it looks long, but I tried to separate it stage and install things using the following priority:

- Arch repos using `pacman`
- AUR repos using `yay`
- PyPi using `pip`
- GitHub repos, just cloning and using `pip`

Almost all dependencies are from the main repo. Here you can see how I installed `rar` (from AUR).
I have to run the tests now, but the dependencies seems to be solved in a clean way.
I'm currently installing KiBoM, PcbDraw and iBoM from  the repos manually, I'll see if I can install KiBoM and iBoM using AUR. 
PcbDraw is a problem, in particular with the 1.0.0 released today.

**Reply by set-soft** [reply]
I did a couple of adjusts to the above script:

1. Arch/Manjaro seems to have strong problems with the minimum set of fonts needed to make the installed tools usable. For ImageMagick you must be sure you install `gsfonts` otherwise you can't handle text because the primitives uses Helvetica (an ancient sans serif font found even on Commodore 128 programs) and Manjaro doesn't install any replacement as ImageMagick dependency. Installing it ImageMagick can draw text, but I'm getting warnings like this:
```
Pango-WARNING **: 15:59:56.732: failed to create cairo scaled font, expect ugly output. the offending font is 'Nimbus Roman, 1333333'
Pango-WARNING **: 15:59:56.734: font_face status is: error occurred in libfreetype
```

2. Installing PcbDraw from the repo isn't trivial, you must tell git to also include the submodules, in addition to the right branch. I updated the script in the repo: https://github.com/INTI-CMNB/KiBot/blob/master/tools/arch_install.sh

Installing all the needed tools for testing:

```shell
sudo pacman -S flake8 python-pytest python-pytest-xdist python-wheel diffutils fluxbox x11vnc wmctrl unzip zbar python-coverage wget --noconfirm
# sudo pacman procps-ng python-setuptools python-pip curl --noconfirm
# For the makefile
sudo pacman -S --noconfirm  make python-pre-commit
yay -S --noconfirm  python-xlsx2csv
```

I was able to run all the regression tests. Some tests files because KiCad image generation is slightly inconsistent, changes in the KiCad version and/or underlying libs produces small differences, specially in the gradient found at edges. But overall KiBot seems to work as expected using the Manjaro image.

**Reply by taotieren** [reply]
```bash
yay -S kibot
```

**Comment by mdeweerd** [comment]
kibot-checker is/has been usefull, I would even add it as a kibot target to log the version in a file in the target directory to be able to check later which versions were used in producing the output.

**Reply by set-soft** [reply]
I added a new output called `info`, it just saves the `kibot-check` output.
More info could be added in the future.

**Comment by erikbozelie** [comment]
upgrading kiauto to V2.0 and pcbdraw to version 0.9.0.3 has caused some other issues.

Ill try to make a docker image for you.
